ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt

```diff
@@ -124,64 +126,46 @@ fun PhotobookPage(appState: AppState) {
     val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
 
     LaunchedEffect(Unit) {
-        try {
-            val response = PhotoApiService.getAllUserPhotos(userid, appState.dataStore.getJwt())
-            if (response.status == HttpStatusCode.OK) {
-                val listResponse: ListResponse<PhotoResponse> = response.body()
-                val photoList = listResponse.list.map { photoRes ->
-                    val date = LocalDate.of(photoRes.year.toInt(), photoRes.month.toInt(), photoRes.day.toInt())
-                    val datestr = "${date.dayOfMonth} ${
-                        capitalize(date.dayOfWeek.toString().take(3))
-                    }"
-                    PhotobookPhoto(
-                        datestr, decodeImage(photoRes.photoBase64)
-                    )
-                }
-                appState.photos.clear()
-                appState.photos.addAll(photoList.reversed())
-            } else {
-                val statusResponse: StatusResponse = response.body()
-                println("getting photolist failed")
-                println(statusResponse.body)
-                // TODO:handle failed image creation
-                appState.pageState.value = PageStates.HOME
-            }
-        } catch (e: Exception) {
-            // TODO: handle error
-            println("some error occured while fetching photolist")
-            println(e)
+        val photos = appState.userStrategy!!.getAllPhotos(appState)
+        val photoList = photos?.map { photoRes ->
+            val date =
+                LocalDate.of(photoRes.year, photoRes.month, photoRes.day)
+            val datestr = "${date.dayOfMonth} ${
+                capitalize(date.dayOfWeek.toString().take(3))
+            }"
+            PhotobookPhoto(
+                datestr, decodeImage(photoRes.photoBase64)
+            )
+        }
+        appState.photos.clear()
```

Comment:
for my understanding, why are we clearing here?

Reply:
It's a failsafe, we clear all existing photos from the state and reload it from mongo. My thoughts were that if the user had an app on another phone and made changes to the photolist, we can wipe the slate and load it again, instead of appending to the existing list and dealing with duplicates.